homework_7/animals/Animal.java

```diff
@@ -0,0 +1,58 @@
+package animals;
+
+public class Animal {
+    private String nameAnimal;
+    private static int animalsCount;
+    private static int catsCount;
+    private static int dogsCount;
+
+    public Animal(String nameAnimal) {
+        this.nameAnimal = nameAnimal;
+        animalsCount++;
+        if (this instanceof Cat) {
```

Comment:
Class Animal shouldn't know any details about any subclass! This is a bad idea to implement counters this way. Please remove this logic from here